deployment/terraform/create.go

```diff
@@ -160,22 +174,69 @@ func (t *Terraform) Create() error {
 
 			// Starting mattermost.
 			mlog.Info("Starting mattermost", mlog.String("host", ip))
-			cmd := fmt.Sprintf("sudo service mattermost start")
+			cmd := "sudo service mattermost start"
 			if err := sshc.RunCommand(cmd); err != nil {
 				mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.Err(err))
 				return
 			}
 		}()
 	}
+}
 
-	// TODO: display the entire cluster info from terraformOutput later
-	// when we have cluster support.
-	mlog.Info("Deployment complete.")
+func (t *Terraform) setupProxyServer(output *terraformOutput, extAgent *ssh.ExtAgent) {
+	if len(output.ProxyIP.Value) == 0 {
+		mlog.Error("no proxy server ip found in terraform output. proxy server will not be configured")
+		return
+	}
+	ip := output.ProxyIP.Value[0]
+	sshc, err := extAgent.NewClient(ip)
+	if err != nil {
+		mlog.Error("error in getting ssh connection", mlog.String("ip", ip), mlog.Err(err))
+		return
+	}
+	func() {
+		defer func() {
+			err := sshc.Close()
+			if err != nil {
+				mlog.Error("error closing ssh connection", mlog.Err(err))
+			}
+		}()
 
-	return nil
+		// Upload service file
+		mlog.Info("Uploading nginx config", mlog.String("host", ip))
+
+		backends := ""
+		for _, addr := range output.Instances.Value {
+			backends += "server " + addr.PrivateIP + ":8065;\n"
+		}
+
+		files := []struct {
+			path    string
+			content string
+		}{
+			{content: strings.TrimSpace(fmt.Sprintf(nginxSiteConfig, backends)), path: "/etc/nginx/sites-available/mattermost"},
+			{content: strings.TrimSpace(sysctlConfig), path: "/etc/sysctl.conf"},
+			{content: strings.TrimSpace(nginxConfig), path: "/etc/nginx/nginx.conf"},
+			{content: strings.TrimSpace(limitsConfig), path: "/etc/security/limits.conf"},
+		}
+		for _, fileInfo := range files {
+			rdr := strings.NewReader(fileInfo.content)
+			if err := sshc.Upload(rdr, fileInfo.path, true); err != nil {
+				mlog.Error("error uploading file", mlog.Err(err), mlog.String("file", fileInfo.path))
+				return
+			}
+		}
+
+		cmd := "sudo shutdown -r now &"
```

Comment:
I am wondering whether a restart is actually needed here. The nginx things can be done with an nginx restart, and the sysctl things can be done with `sudo sysctl -p`. Anything else we are missing ?